contrib/startup_regtest.sh

```diff
@@ -153,39 +153,58 @@ start_ln() {
 
 fund_ln() {
 
+	if [ -z "$1" ]; then
+		node1=1
+	else
+		node1="$1"
+	fi
+
+	if [ -z "$2" ]; then
+		node2=2
+	else
+		node2="$2"
+	fi
+
 	ADDRESS=`bitcoin-cli -regtest getnewaddress`
+
+	echo "minning into address " $ADDRESS
+
 	bitcoin-cli -regtest generatetoaddress 50 $ADDRESS
 
-	echo $ADDRESS
+	echo "Mined into $ADDRESS, checking balance"
 	bitcoin-cli -regtest getbalance
 
-	L2_NODE_ID=`$LCLI --lightning-dir=/tmp/l2-regtest getinfo | jq -r .id`
-	L2_NODE_PORT=`$LCLI --lightning-dir=/tmp/l2-regtest getinfo | jq .binding[0].port`
+	L2_NODE_ID=`$LCLI --lightning-dir=/tmp/l$node2-regtest getinfo | jq -r .id`
+	L2_NODE_PORT=`$LCLI --lightning-dir=/tmp/l$node2-regtest getinfo | jq .binding[0].port`
```

Comment:
Prefer $LCLI -F here, so you can avoid requiring jq?

Reply:
What's the ideal way to parse the -F output -- is using sed appropriate here?

`L2_NODE_ID=$($LCLI -F --lightning-dir=/tmp/l$node2-regtest getinfo | sed -n 's/^id=\(.*\)/\1/p')`

eval almost works but it doesn't like the array notation.

```
$ eval $($LCLI -F --lightning-dir=/tmp/l$node2-regtest getinfo)
-bash: binding[0].type=ipv4: command not found
```

Reply:
I went with using `sed`